src/main/java/org/jabref/gui/customizefields/CustomizeGeneralFieldsDialogViewModel.java

```diff
@@ -43,48 +46,36 @@ public StringProperty fieldsTextProperty() {
     }
 
     public void saveFields() {
+        Map<String, Set<Field>> customTabsMap = new LinkedHashMap<>();
         String[] lines = fieldsText.get().split("\n");
-        int i = 0;
-        for (; i < lines.length; i++) {
-            String[] parts = lines[i].split(":");
+
+        for (String line : lines) {
+            String[] parts = line.split(":");
             if (parts.length != 2) {
-                // Report error and exit.
-                String field = Localization.lang("field");
-                String title = Localization.lang("Error");
-                String content = Localization.lang("Each line must be of the following form") + " '" +
-                                 Localization.lang("Tabname") + ':' + field + "1;" + field + "2;...;" + field + "N'";
-                dialogService.showInformationDialogAndWait(title, content);
+                dialogService.showInformationDialogAndWait(
+                        Localization.lang("Error"),
+                        Localization.lang("Each line must be of the following form: 'tab:field1;field2;...;fieldN'."));
                 return;
             }
 
             String testString = BibtexKeyGenerator.cleanKey(parts[1], preferences.getEnforceLegalKeys());
             if (!testString.equals(parts[1]) || (parts[1].indexOf('&') >= 0)) {
-                String title = Localization.lang("Error");
-                String content = Localization.lang("Field names are not allowed to contain white space or the following "
-                                                   + "characters")
-                                 + ": # { } ( ) ~ , ^ & - \" ' ` ʹ \\";
-                dialogService.showInformationDialogAndWait(title, content);
+                dialogService.showInformationDialogAndWait(
+                        Localization.lang("Error"),
+                        Localization.lang("Field names are not allowed to contain white spaces or certain characters (%0).",
+                                "# { } ( ) ~ , ^ & - \" ' ` ʹ \\"));
                 return;
             }
-            preferences.setCustomTabsNameAndFields(parts[0], parts[1], i);
 
+            customTabsMap.put(parts[0], FieldFactory.parseFieldList(parts[1]));
         }
-        preferences.purgeSeries(JabRefPreferences.CUSTOM_TAB_NAME, i);
-        preferences.purgeSeries(JabRefPreferences.CUSTOM_TAB_FIELDS, i);
+
+        preferences.storeEntryEditorTabList(customTabsMap);
```

Comment:
Shouldn't `updateEntryEditorTabList` be automatically called if `storeEntryEditorTabList` is called?